crates/oxc_transformer/src/proposals/explicit_resource_management.rs

```diff
@@ -0,0 +1,768 @@
+//! Proposal: Explicit Resource Management
+//!
+//! This plugin transforms explicit resource management syntax into a series of try-catch-finally blocks.
+//!
+//! ## Example
+//!
+//! Input:
+//! ```js
+//! for await (using x of y) {
+//!     doSomethingWith();
+//! }
+//! ```
+//!
+//! Output:
+//! ```js
+//! for await (const _x of y)
+//! try {
+//!     var _usingCtx = babelHelpers.usingCtx();
+//!     const x = _usingCtx.u(_x);
+//!     doSomethingWith(x);
+//! } catch (_) {
+//!     _usingCtx.e = _;
+//! } finally {
+//!     _usingCtx.d();
+//! }
+//! ```
+//!
+//! ## Implementation
+//!
+//! Implementation based on [@babel/babel-plugin-syntax-explicit-resource-management](https://babeljs.io/docs/babel-plugin-syntax-explicit-resource-management).
+//!
+//! ## References:
+//! * Babel plugin implementation: <https://github.com/babel/babel/tree/v7.26.9/packages/babel-plugin-syntax-explicit-resource-management>
+//! * Explicit Resource Management TC39 proposal: <https://github.com/tc39/proposal-explicit-resource-management>
+
+use std::mem;
+
+use oxc_allocator::Address;
+use oxc_allocator::GetAddress;
+use oxc_ast::ast::BlockStatement;
+use oxc_ast::ast::Class;
+use oxc_ast::ast::ClassType;
+use oxc_ast::{
+    NONE,
+    ast::{
+        BindingPatternKind, CatchClause, Declaration, ExportDefaultDeclarationKind, Expression,
+        ForStatementLeft, ImportOrExportKind, ModuleExportName, Program, SimpleAssignmentTarget,
+        Statement, VariableDeclarationKind,
+    },
+};
+use oxc_ecmascript::BoundNames;
+use oxc_semantic::ScopeId;
+use oxc_semantic::{ScopeFlags, SymbolFlags};
+use oxc_span::{Atom, SPAN};
+use oxc_traverse::{BoundIdentifier, Traverse};
```

Comment:
Style nits:

For brevity, please combine `use` statements, so there's only 1 for each crate.

For `oxc_ast`, just blanket import all the AST types with `use oxc_ast::ast::*;`.

Throughout the visitors, please just `node: &mut ForOfStatement<'a>` instead of `node: &mut oxc_ast::ast::ForOfStatement<'a>`.

Ditto for `TraverseCtx` - import it at top of file, to avoid repetition of `&mut oxc_traverse::TraverseCtx<'a>`.

Ditto `Vec`. Please import at the top with `use oxc_allocator::Vec as ArenaVec;`. That's our naming convention in transformer to make it clear what's an arena `Vec` and what's a `std::vec::Vec`.

Reply:
makes sense. should be fixed now - thanks